File: NNLT25/bai-34/bai-34.cpp
#include <bits/stdc++.h>
/* backtracking */
using namespace std;

string num_max(string N, int K) {
    string S = "";
    for (char digit : N) {
        while (!S.empty() && K > 0 && S.back() < digit) {
            S.pop_back();
            K--;
        }
        S.push_back(digit);
    }
    while (K > 0) {
        S.pop_back();
        K--;
    }
    return S;
}

int main() {
    string N;
    int K;
    cin >> N >> K;
    cout << num_max(N, K);
}

File: test-archive/NNLT25/bai-34/bai-34.py
def num_max(N, K):
    a = list(str(N))
    S = []
    
    for digit in a:
        while S and K > 0 and S[-1] < digit:
            S.pop()
            K -= 1
        S.append(digit)
    
    while K > 0:
        S.pop()
        K -= 1
    
    return ''.join(S)

N = int(input())
K = int(input())
print(num_max(N, K))
